File: server/server-bukkit/src/main/java/server/bukkit/util/font/LengthyComponent.java
package server.bukkit.util.font;


import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.TextDecoration;

public class LengthyComponent {

    private static final LengthyComponent EMPTY = new LengthyComponent(Component.empty(), 0);

    private final Component symbol;
    private final int length;

    public LengthyComponent(Component symbol, int length) {
        if (symbol != Component.empty() && symbol.style().decoration(TextDecoration.BOLD) == TextDecoration.State.NOT_SET) {
            throw new IllegalArgumentException("invalid bold state");
        }
        this.symbol = symbol;
        this.length = length;
    }

    public Component getSymbol() {
        return symbol;
    }

    public int getLength() {
        return length;
    }

    public static LengthyComponent empty() {
        return EMPTY;
    }

}

File: server/server-bukkit/src/main/kotlin/server/bukkit/util/text/LengthyComponent.kt
package server.bukkit.util.text

import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.TextDecoration


class LengthyComponent(val symbol: Component, val length: Int) {

    init {
        require(isEmpty() || isBoldSet()) { "invalid bold state" }
    }

    fun isEmpty(): Boolean {
        return symbol == Component.empty()
    }

    fun isBoldSet(): Boolean {
        return symbol.style().decoration(TextDecoration.BOLD) != TextDecoration.State.NOT_SET
    }

    companion object {
        val EMPTY = LengthyComponent(Component.empty(), 0)
    }
}
